src/deepsparse/server/config.py

```diff
@@ -12,152 +12,151 @@
+    image_sizes: List[Tuple[int, int]] = Field(
+        description="The list of image sizes the model should accept"
     )
-    workers: str = Field(
-        default=max(1, cpu_architecture().num_available_physical_cores // 2),
+
+
+class EndpointConfig(BaseModel):
+    name: Optional[str] = Field(
+        default=None,
         description=(
-            "The number of maximum workers to use for processing pipeline requests. "
-            "Defaults to the number of physical cores on the device."
+            "Name of the model used for logging & metric purposes. "
+            "If not specified 'endpoint-<index>' will be used."
         ),
     )
-    integration: str = Field(
-        default="default",
-        description=(
-            "Name of deployment integration that this server will be deployed to "
-            "Currently supported options are None for default inference and "
-            "'sagemaker' for inference deployment with AWS Sagemaker"
-        ),
+
+    endpoint: Optional[str] = Field(
+        default=None,
+        description="Optional url to use for this endpoint. E.g. '/predict'. "
+        "If not specified '/endpoint-<index>/predict' will be used",
     )
 
+    task: str = Field(description="Task this endpoint performs")
 
-@lru_cache()
-def server_config_from_env(env_key: str = ENV_DEEPSPARSE_SERVER_CONFIG):
-    """
-    Load a server configuration from the targeted environment variable given by env_key.
+    model: str = Field(description="Location of the underlying model to use.")
 
-    :param env_key: the environment variable to load the configuration from.
-        Defaults to ENV_DEEPSPARSE_SERVER_CONFIG
-    :return: the loaded configuration file
-    """
-    config_file = os.environ[env_key]
+    batch_size: int = Field(
+        default=1, description="The batch size to compile the model for."
+    )
 
-    if not config_file:
-        raise ValueError(
-            "environment variable for deepsparse server config not found at "
-            f"{env_key}"
-        )
+    bucketing: Optional[Union[ImageSizesConfig, SequenceLengthsConfig]] = Field(
+        default=None,
+        description=(
+            "What input shapes this model can accept."
+            "Example for multiple sequence lengths in yaml: "
+            "```yaml\n"
+            "bucketing:\n"
+            "  sequence_lengths: [16, 32, 64]\n"
+            "```\n"
+        ),
+    )
 
-    if config_file.startswith(ENV_SINGLE_PREFIX):
-        config_dict = json.loads(config_file.replace(ENV_SINGLE_PREFIX, ""))
-        config = ServerConfig()
-        config.models.append(
-            PipelineConfig(
-                task=config_dict["task"],
-                model_path=config_dict["model_path"],
-                batch_size=config_dict["batch_size"],
-            )
+    def to_pipeline_config(self) -> PipelineConfig:
+        input_shapes, kwargs = _unpack_bucketing(self.task, self.bucketing)
+
+        return PipelineConfig(
```

Comment:
we should set the pipeline `alias` to the `name` or `endpoint` value here, that way the logger can have something easy to log the pipeline name as